Shared/Samples/Change viewpoint/ChangeViewpointView.swift

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct ChangeViewpointView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    @State private var visibleArea: ArcGIS.Polygon?
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            VStack {
+                MapView(map: model.map, graphicsOverlays: [model.graphicsOverlays])
+                    .onScaleChanged { model.scale = $0 }
+                    .onVisibleAreaChanged { visibleArea = $0 }
+                    .onChange(of: $model.viewpointType.wrappedValue) { type in
+                        switch type {
+                        case .centerAndScale:
+                            Task {
+                                try await mapViewProxy.setViewpointCenter(model.londonCoordinate, scale: 40000)
+                            }
+                        case .geometry:
+                            Task {
+                                try await mapViewProxy.setViewpointGeometry(
+                                    model.griffithParkGeometry,
+                                    padding: 50
+                                )
+                            }
+                        case .animate:
+                            Task {
+                                if let center = visibleArea?.extent.center {
+                                    try await mapViewProxy.setViewpoint(
+                                        Viewpoint(center: center, scale: model.scale / 2),
+                                        duration: 5
+                                    )
+                                }
+                            }
+                        }
+                    }
+                HStack {
```

Comment:
For samples, we typically use a toolbar modifier, so that SwiftUI can reorganize the UI on a Mac more easily. Up to you which UI you prefer.